[CLU] Edited ReadMe.md (#24115)
Azure/azure-sdk-for-net: sdk/cognitivelanguage/Azure.AI.Language.Conversations/README.md

# Azure Cognitive Language Services Conversations client library for .NET

> Conversations (aka LuisVNext) is a cloud-based conversational AI service that applies custom machine-learning intelligence to a user's conversational, natural language text to predict overall meaning, and pull out relevant, detailed information. The service utilizes state-of-the-art technology to create and utilize natively multilingual models, which means that users would be able to train their models in one language but predict in others. 

Reviewer: "LuisVNext" not only seems formatted incorrect ("LUIS vNext"?) but a term customers probably aren't familiar with nor would they need to know. Maybe instead something like, "Azure Conversations - the new version of Language Understanding (LUIS) - is a cloud-based ..."

Should also say "pulls out ..." for word agreement with "applies custom ...".

```suggestion
Azure Conversations - the new version of Language Understanding (LUIS) - is a cloud-based conversational AI service that applies custom machine-learning intelligence to a user's conversational, natural language text to predict overall meaning; and pulls out relevant, detailed information. The service utilizes state-of-the-art technology to create and utilize natively multilingual models, which means that users would be able to train their models in one language but predict in others. 
```

[Source code][conversationanalysis_client_src] | [Package (NuGet)][conversationanalysis_nuget_package] | [API reference documentation][conversationanalysis_refdocs] | [Product documentation][conversationanalysis_docs] | [Samples][conversationanalysis_samples]

## Getting started

### Install the package

Install the Azure Cognitive Language Services Conversations client library for .NET with [NuGet][nuget]:

```powershell
dotnet add package Azure.AI.Language.Conversations --prerelease
```

### Prerequisites

* An [Azure subscription][azure_subscription]

> * An existing Text Analytics resource

Reviewer: Is "Text Analytics" here right? Is it the same resource type? At the very least, we may want to point that out, but I believe they are different resource types.

Author: When I use the CLU Language Studio (the one in preview) to create new projects, I need to choose an Azure language resource of type "Text Analytics". I am not sure if this will still be the case after preview is over, I'll check and let you know.

Author: I checked with a service team privately, and it seems like this will remain to be the case.

> Note: the new unified Cognitive Language Services are not currently available for deployment.

### Authenticate the client

In order to interact with the Conversations service, you'll need to create an instance of the [`ConversationAnalysisClient`][conversationanalysis_client_class] class. You will need an **endpoint**, and an **API key** to instantiate a client object. For more information regarding authenticating with Cognitive Services, see [Authenticate requests to Azure Cognitive Services][cognitive_auth].

#### Get an API key

You can get the **endpoint** and an **API key** from the Cognitive Services resource in the [Azure Portal][azure_portal].

Alternatively, use the [Azure CLI][azure_cli] command shown below to get the API key from the Cognitive Service resource.

```powershell
az cognitiveservices account keys list --resource-group <resource-group-name> --name <resource-name>
```

#### Create a ConversationAnalysisClient

Once you've determined your **endpoint** and **API key** you can instantiate a `ConversationAnalysisClient`:

> ```C# Snippet:ConversationAnalysisClient_Create

Reviewer: These code changes actually need to be made in the tests/Azure.AI.Language.Conversations.Tests/samples directory, then you run (rrom the repo root) `eng/scripts/CodeChecks.ps1 cognitivelanguage` to update them and make sure everything is correct. That's why this PR build is also failing.

Reviewer: We do this to make sure 1) our samples actually compile, and 2) when recorded or run live, the samples actually work like we expect.

Reviewer: ...or are these from my previous changes?

Uri endpoint = new Uri("https://myaccount.api.cognitive.microsoft.com");
AzureKeyCredential credential = new AzureKeyCredential("{api-key}");

ConversationAnalysisClient client = new ConversationAnalysisClient(endpoint, credential);
```

## Key concepts

### ConversationAnalysisClient

> The [`ConversationAnalysisClient`][conversationanalysis_client_class] is the primary interface for making predictions using your deployed Conversation models. It provides both synchronous and asynchronous APIs to submit queries.

Reviewer: Nit (consistency): 
```suggestion
The [`ConversationAnalysisClient`][conversationanalysis_client_class] is the primary interface for making predictions using your deployed Conversations models. It provides both synchronous and asynchronous APIs to submit queries.
```

## Examples

The Azure.AI.Language.Conversations client library provides both synchronous and asynchronous APIs.

The following examples show common scenarios using the `client` [created above](#Create a ConversationAnalysisClient).

> ### Analyze a Conversation

Reviewer: ```suggestion
### Analyze a conversation
```

> To analyze a conversation, we can then simply call the `client.AnalyzeConversation()` method, which takes the project name, deployment name and query as parameters.

Reviewer: ```suggestion
To analyze a conversation, we can then call the `client.AnalyzeConversation()` method which takes the project name, deployment name, and query as parameters.
```

```C# Snippet:ConversationAnalysis_AnalyzeConversation
Response<AnalyzeConversationResult> response = client.AnalyzeConversation(
    "Menu",
    "production",
    "We'll have 2 plates of seared salmon nigiri.");

Console.WriteLine($"Top intent: {response.Value.Prediction.TopIntent}");
```

> To set additional properties such as language of the query and verbosity, you can initialise an `AnalyzeConversationOptions` instance with your chosen parameters. You can then call `AnalyzeConversation()` using the options as a parameter.

Reviewer: ```suggestion
To set additional properties such as language of the query and verbosity, you can initialize an `AnalyzeConversationOptions` instance with your chosen parameters. You can then call `AnalyzeConversation()` using the options as a parameter.
```

You will notice that additional information is logged, like the client request ID of the operation.

> ```text

Reviewer: This should be kept. It's standard in all our SDK READMEs.

Author: The original message is about the "azure search service". I wanted to recreate the error for the conversations service using the `BadRequest` Test in Readme.cs. But I received a 401 error instead. I will bring back the original message.

Reviewer: You could generate an error and copy that out.

[contributing]: https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md
[core_logging]: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/samples/Diagnostics.md
[nuget]: https://www.nuget.org/

> <!--References after mergin to main repo-->

Reviewer: Comment is unnecessary (and misspelled). Our linker checker takes into account if a PR's links *will* be in main after merging.

[conversationanalysis_client_class]: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/cognitivelanguage/Azure.AI.Language.Conversations/src/ConversationAnalysisClient.cs
[conversationanalysis_client_src]: https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/cognitivelanguage/Azure.AI.Language.Conversations/src/
[conversationanalysis_samples]: https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/cognitivelanguage/Azure.AI.Language.Conversations/samples/

<!--Update once nuget is released-->
[conversationanalysis_nuget_package]: https://nuget.org/packages

> <!--Will be updated once service documentation is public-->

Reviewer: Let's open tracking issues for these. Be sure to add them to the "Cognitive Services - Language" label.

Author: Unfortunately, I can't add any labels to issues :(. Does this also have to do with the write access that I don't have?

Here is a link to the issue. Can you please add the label ? [https://github.com/Azure/azure-sdk-for-net/issues/24163]( https://github.com/Azure/azure-sdk-for-net/issues/24163)
